daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -521,6 +525,44 @@ class Engine(val config: EngineConfig = Engine.StableConfig) {
     } yield ()
   }
 
+  /** Given a contract argument of the given template id, calculate the interface
+    * view of that API.
+    */
+  def computeInterfaceView(
+      templateId: Identifier,
+      argument: Value,
```

Comment:
Any reason why argument is not versioned ?

Reply:
So currently, I don’t make the assumption that the argument is typechecked & normalized. We could make that assumption but I think it actually buys us very little. We need to convert to `SValue` anyway and doing that via `translateValue` which does normalization & typechecking seems easy and not significantly more expensive and it reduces the assumptions we have to make about inputs.

Reply:
> not significantly more expensive and it reduces the assumptions we have to make about inputs.

If we implement #13865, deserializing a normalized value can be significantly faster.

Reply:
Ah I see, happy to change this in follow up PRs if you want. What function would I use in that case instead of `translateValue`?